src/config/src/flagsmith.ts

```diff
@@ -0,0 +1,118 @@
+import { Flag } from './flag'
+import { INITIALIZE_WAIT_TIMEOUT, MINIMUM_UPDATE_COOLDOWN } from './constants'
+import { Store } from './store'
+
+export class Flagsmith {
+  private lastUpdated: number
+  private flags: Flag[] = []
+  private initialized = false
+  private readonly core: { name: string; enabled: boolean }[] = [
+    {
+      name: 'vscode',
+      enabled: true,
+    },
+  ]
+  constructor(
+    private readonly config: { environmentKey: string; flagsmithAPI: string },
+    private readonly traits: { [name: string]: string },
+    private readonly storage: Store,
+  ) {
+    this.lastUpdated = 0
+    for (const flag of this.core) {
+      this.storage.has(flag.name)
+        ? this.flags.push(
+            new Flag(flag.name, flag.enabled, this.storage.get(flag.name)),
+          )
+        : this.flags.push(new Flag(flag.name, flag.enabled))
+      this.storage.put(flag.name, this.flags[this.flags.length - 1].enabled)
+    }
+    this.loadFlags().then(() => (this.initialized = true))
+  }
+  async get(flagName: string) {
+    await this.loadFlags()
+    for (const flag of this.flags) {
+      if (flagName === flag.name) {
+        return flag
+      }
+    }
+    // @todo log somewhere
+    return new Flag(flagName, false, false, false, false)
+  }
+  private async loadFlags() {
```

Comment:
Decent amount going on here. Are there any behaviors here we can test?

Reply:
Looks like there are network calls here that might make testing more difficult and I don't really care to test networking rn. Potentially could test the second half if that half is cohesive as its own function

Reply:
I'm happy to refactor that, would just due to size reasons not want to do the testing and refactoring here.